addons/dynamishelper/dynamishelper.lua

```diff
@@ -24,408 +24,427 @@
-    			windower.send_command('ll profile dynamis-'..player..'.txt')
-   			else print("Invalid light luggage option.")
-   			end
-  	 elseif params[1]:lower() == "proc" then
-   			if params[2]:lower() == "on" then
-   				proc = params[2]
-   				print('Proc feature enabled.')
-   			elseif params[2]:lower() == "off" then
-   		 		proc = params[2]
-    			windower.text.set_visibility('proc_box',false)
-    			print('Proc feature disabled.')
-    		elseif params[2]:lower() == "pos" then
-   				pposx, pposy = tonumber(params[3]), tonumber(params[4])
-   				initializeproc()
-   			end
-		end
-	end
-end)
-
-
+-------------------------------------------------------------------------------
+-- Parse the chat messages for drops, staggers and time extensions ------------
+-------------------------------------------------------------------------------
 windower.register_event('incoming text',function (original, new, color)
---	print('event_incoming_text function')
-	if timer == 'on' then
-  		a,b,fiend = string.find(original,"%w+'s attack staggers the (%w+)%!")
-   		if fiend == 'fiend' then
-			StaggerCount = StaggerCount + 1
-    		windower.send_command('timers c '..StaggerCount..' 30 down')
-    		return new, color
-    	end
-	end
- 	if tracker == 'on' then
-     	a,b,item = string.find(original,"%w+ obtains an? ..(%w+ %w+ %w+ %w+)..\46")
-     		if item == nil then
-      	 		a,b,item = string.find(original,"%w+ obtains an? ..(%w+ %w+ %w+)..\46")
-       				if item == nil then
-         				a,b,item = string.find(original,"%w+ obtains an? ..(%w+%-%w+ %w+)..\46")
-          					if item == nil then
-           						a,b,item = string.find(original,"%w+ obtains an? ..(%w+ %w+)..\46")
-         					end
-       				end
-     		end
--- 		a,b,item = string.find(original,"%w+ obtains an? ..(.*)..\46")
- 		if item ~= nil then
- 			item = item:lower()
- 			for i=1, #Currency do
-				if item == Currency[i]:lower() then
-   					Currency[Currency[i]] = Currency[Currency[i]] + 1
-   				end
- 			end
- 			obtainedf()
- 		end
-    	initializebox()
- 	end
- 	return new, color
-end)
-
-function obtainedf()
-	obtained = nil
- 	for i=1,#Currency do
- 		if Currency[Currency[i]] ~= 0 then
-  			if obtained == nil then
-  				obtained = " "
-  			end
-   			obtained = (obtained..Currency[i]..': '..Currency[Currency[i]]..' \n ')
- 		end
- 	end
-end
-
-windower.register_event('zone change', function(id)
-	goodzone = ProcZones:contains(id)
-	if not goodzone then
-		windower.text.set_visibility('proc_box', false)
-	end
+    local time = nil
+    local item = nil
+    local fiend = original:match("%w+'s attack staggers the (%w+)%!")
+    if settings.timer then
+        if fiend == 'fiend' then
+            windower.send_command('timers c "'..current_mob..'" 30 down stun')
+            return new, color
+        end
+    end
+    if string.find(original,"remaining in Dynamis.") then
```

Comment:
There are just a few inconsistencies in method calls. The original addon uses `string.find(original, 'something')` whereas you seem to use `original:match('something')`, but left the old code in the old style, which results in mixed invocation styles within the file. I would recommend to use the latter style for every call, but I'll leave this up to you.

Style aside, I would suggest to match the strings as closely as possible. This would also match if a player wrote "Not much longer remaining in Dynamis. We should hurry.". I would check the original mode (called `color` here) as well as matching the text as tightly as possible. I don't know the exact message, but iirc it goes something like "30 minutes remaining in Dynamis.", which I would parse with `original:endswith("%d+ minutes remaining in Dynamis.")` or something like that. If you need to alternate between `minutes` and `hours` you can use our regex matching library (not sure if the game reports hours as well, the code afterwards suggests it does not, just mentioned it in case I'm not remembering it correctly).

This goes for the next few `string.find` checks as well.